### PR TITLE
Testing FedAvgGrad

### PR DESCRIPTION
**Pull request by danielspg**
Created a Strategy named FedAvgGrad:

the FedAvgGrad follows the following sample structure:
FedAvgGrad

Created a example which utilizes the FedAvgGrad strategy:
A test run is shown below:
FedAvgGradTest

### CODE REVIEW

**Review by pedropgusmao — COMMENTED** [review]
Hello @danielspg  Thanks for your contribution. Would mind include the tests for this strategy as well? You can use https://github.com/adap/flower/blob/main/src/py/flwr/server/strategy/fedavg_test.py for reference .

**Comment by danielspg** [comment]
1. Modified [FedAvgGrad](https://github.com/danielspg/flower/blob/testing_FedAvgGrad/src/py/flwr/server/strategy/FedAvgGrad.py) strategy code to accommodate DEPRECATIONS of Strategy.aggregate_fit(...) and some other functions in Flower.

2. Fixed the accuracy issue in example: [pytorch_mnist_FedAvgGrad](https://github.com/danielspg/flower/tree/testing_FedAvgGrad/examples/pytorch_mnist_FedAvgGrad)
Below is the execution of example: Using 2 clients and training for 5 round. Final accuracy was around 97%
MNIST_FedAvgGrad


3. **Added Example utilizing FedAvgGrad strategy based on CIFAR dataset: [quickstart_pytorch_FedAvgGrad](https://github.com/danielspg/flower/tree/testing_FedAvgGrad/examples/quickstart_pytorch_FedAvgGrad)**
Below is the execution of example: Using 2 clients and training for 3 rounds. Final Accuracy was around 43%
Cifar_FedAvgGrad

**Comment by jafermarq** [comment]
Hi @danielspg, sorry for the long delay with your PR. We are still happy to move ahead with your contribution but we were thinking to start with just the `FedAvgGrad` strategy. This could be the building block for subsequent strategies that operate with gradients (understood as the difference between the global model received by clients and the locally update model) as opposed to models.

Are you still interested in integrating `FedAvgGrad` into Flower?
